18F/identity-idp
# LG-10177: Always show Face / Touch Unlock in list of configured MFAs

## 🎫 Ticket

[LG-10177](https://cm-jira.usa.gov/browse/LG-10177)

## 🛠 Summary of changes

Removes behavior which would prevent a user from attempting to authenticate an existing WebAuthn configuration on a device which does not have a platform authenticator available ([`isUserVerifyingPlatformAuthenticatorAvailable`](https://developer.mozilla.org/en-US/docs/Web/API/PublicKeyCredential/isUserVerifyingPlatformAuthenticatorAvailable_static)).

**Why?** Because the user's credential may be accessible via cloud services regardless of the availability of a local platform authenticator. For example, a MacBook laptop that does not have a thumbprint reader may still be able to use an iCloud-synced WebAuthn credential.

## 📜 Testing Plan

1. On a supported device (iOS 16+, Android), go to http://localhost:3000
2. Create an account
3. Select "Face or Touch Unlock" as your MFA method
4. Continue account creation
5. On a device which does not have a platform authenticator (e.g. laptop without thumbprint reader, maybe a virtual machine, or Firefox browser), go to http://localhost:3000
6. Sign in to the account created between steps 2-4
7. Observe that you are not redirected to an error page when clicking "Use Face or Touch Unlock"
8. Click "Choose another method"
9. Observe that the "Face or Touch Unlock" option is included in the list of configured MFA methods

## 👀 Screenshots

Screen|Before|After
---|---|---
Authenticate|![Screen Shot 2023-07-06 at 3 37 17 PM](https://github.com/18F/identity-idp/assets/1779930/a2bf8851-c9ff-4957-8351-bb059c4aba77)|![Screen Shot 2023-07-06 at 3 38 10 PM](https://github.com/18F/identity-idp/assets/1779930/0dc0eaa7-67fc-46ec-8105-27b449bab84a)†
MFA List|![Screen Shot 2023-07-06 at 3 35 18 PM](https://github.com/18F/identity-idp/assets/1779930/8a87d808-4644-49a5-8c65-33c7977ee32c)|![Screen Shot 2023-07-06 at 3 35 11 PM](https://github.com/18F/identity-idp/assets/1779930/43b0e07f-5ded-4745-a67e-802562b05f50)

† Note that the Firefox experience of prompting for Security Key here is far from ideal, though this is not the ideal use-case for platform authenticators, since Firefox does not have any support for Passkeys at all.
